Add rule engine configuration option for changing the implementation of the jq function

kjellwinblad opened the pull request:
This commit adds a rule engine configuration option for changing the implementation module used for the rule engine function jq. The two options are `jq_port` (uses Erlang port programs to interact with jq) and `jq_nif` (uses an Erlang NIF library to interact with jq).

**Things needed before merge:**

* Chinese translation for the description of the config option
* Bump the jq library version after [this PR](https://github.com/emqx/jq/pull/36) has been merged (the config option has no effect before that PR)

terry-xiaoyu commented on apps/emqx_rule_engine/i18n/emqx_rule_engine_schema.conf:
```diff
@@ -261,6 +261,17 @@ of the rule, then the string "undefined" is used.
+    rule_engine_jq_implementation_module {
+      desc {
+          en: "The implementation module for the jq rule engine function. The two options are jq_nif and jq_port. jq_nif uses an Erlang NIF library to access the jq library while jq_port uses an implementation based on Erlang port programs. The jq_nif option (the default option) is the fastest implementation of the two but jq_port is safer as the jq programs will not execute in the same process a ...
+          zh: ""
```
```suggestion
          zh: "jq 规则引擎功能的实现模块。可用的两个选项是 jq_nif 和 jq_ port。jq_nif 使用 Erlang NIF 库访问 jq 库，而 jq_port 使用基于 Erlang Port 的实现。jq_nif 方式（默认选项）是这两个选项中最快的实现，但 jq_port 方式更安全，因为这种情况下 jq 程序不会在 Erlang VM 进程中执行。"
```

terry-xiaoyu commented on apps/emqx_rule_engine/i18n/emqx_rule_engine_schema.conf:
```diff
@@ -261,6 +261,17 @@ of the rule, then the string "undefined" is used.
+          zh: ""
+      }
+      label: {
+          en: "Implementation module used by the rule engine jq function (jq_nif for NIF backend or jq_port for a port backend)"
```
```suggestion
          en: "JQ Implementation Module"
```

terry-xiaoyu commented on apps/emqx_rule_engine/i18n/emqx_rule_engine_schema.conf:
```diff
@@ -261,6 +261,17 @@ of the rule, then the string "undefined" is used.
+      }
+      label: {
+          en: "Implementation module used by the rule engine jq function (jq_nif for NIF backend or jq_port for a port backend)"
+          zh: ""
```
```suggestion
          zh: "JQ 实现模块"
```

kjellwinblad commented:
Updated to jq version with timeout support in the NIF implementation and squashed the previous commits. I think it is ready to be merged now if I get an approving review @terry-xiaoyu .

terry-xiaoyu commented on apps/emqx_rule_engine/i18n/emqx_rule_engine_schema.conf:
```diff
@@ -261,6 +261,17 @@ of the rule, then the string "undefined" is used.
+    rule_engine_jq_implementation_module {
+      desc {
+          en: "The implementation module for the jq rule engine function. The two options are jq_nif and jq_port. With the jq_nif option an Erlang NIF library is used while with the jq_port option an implementation based on Erlang port programs is used. The jq_nif option (the default option) is the fastest implementation of the two but jq_port is safer as the jq programs will not execute in the s ...
+          zh: "jq 规则引擎功能的实现模块。可用的两个选项是 jq_nif 和 jq_ port。jq_nif 使用 Erlang NIF 库访问 jq 库，而 jq_port 使用基于 Erlang Port 的实现。jq_nif 方式（默认选项）是这两个选项中最快的实现，但 jq_port 方式更安全，因为这种情况下 jq 程序不会在 Erlang VM 进程中执行。"
```
change `jq_ port` to `jq_port`

kjellwinblad replied:
Fixed in commit below
